netbox/dcim/models/devices.py

```diff
@@ -1129,3 +1136,88 @@ def delete(self, *args, **kwargs):
             )
 
         return super().delete(*args, **kwargs)
+
+
+class VirtualDeviceContext(NetBoxModel):
+    device = models.ForeignKey(
+        to='Device',
+        on_delete=models.PROTECT,
+        related_name='vdcs',
+        blank=True,
+        null=True
+    )
+    name = models.CharField(
+        max_length=64
+    )
+    status = models.CharField(
+        max_length=50,
+        blank=True,
+        choices=VirtualDeviceContextStatusChoices,
+    )
+    identifier = models.PositiveSmallIntegerField(
+        blank=True,
+        null=True,
+        validators=[MaxValueValidator(255)]
+    )
+    primary_ip4 = models.OneToOneField(
```

Comment:
Do we need primary IPv4/v6 attributes for a VDC? Seems like maybe overkill.

Reply:
So, from my experience (Cisco), we have:

ASA: "Context"
FTD: Instance
Nexus: VDC

In each one, both the Chassis has a IP (Device primary_ip4) as well as each instance/virtual context (VDC: manangement IP, ASA: management IP, FTD: Instance management IP & Diagnostic IP).

Since each have a separate management plane, I think it would be realistic to have the primary IP.